src/components/scenes/SwapProcessingScene.tsx

```diff
@@ -61,6 +65,95 @@ export function SwapProcessingScene(props: Props) {
+
+        let matchingWalletId: string | undefined
+        for (const walletId of Object.keys(account.currencyWallets)) {
+          const wallet = account.currencyWallets[walletId]
+          if (wallet.currencyInfo.pluginId === targetPluginId) {
+            const toAddresses = await wallet.getAddresses({ tokenId: null })
+            const publicAddress = toAddresses[0]?.publicAddress
+            if (
+              fromAddress != null &&
+              publicAddress != null &&
+              fromAddress.toLowerCase() === publicAddress.toLowerCase()
+            ) {
+              matchingWalletId = walletId
+              break
+            }
+          }
+        }
+
+        let finalToWalletId: string = swapRequest.toWallet.id
+        let finalToWallet = swapRequest.toWallet
+        let isWalletCreated = false
+        if (matchingWalletId == null) {
+          // If not found, split from the source chain wallet to the destination
+          // chain wallet type:
+          isWalletCreated = true
+          const splitFromWallet = fromWallet
+          const targetWalletType =
+            account.currencyConfig[targetPluginId]?.currencyInfo.walletType
+          if (targetWalletType == null)
+            throw new Error('Target wallet type unavailable')
+
+          const splitWalletId = await account.splitWalletInfo(
+            splitFromWallet.id,
+            targetWalletType
+          )
+          const newWallet = await account.waitForCurrencyWallet(splitWalletId)
+          finalToWalletId = newWallet.id
+          finalToWallet = newWallet
+        } else {
+          finalToWalletId = matchingWalletId
+          finalToWallet = account.currencyWallets[matchingWalletId]
+        }
+
+        // Navigate back to swap create with the correct wallet selected:
+        navigation.navigate('swapTab', {
+          screen: 'swapCreate',
+          params: {
+            fromWalletId: fromWallet.id,
+            fromTokenId: swapRequest.fromTokenId,
+            toWalletId: finalToWalletId,
+            toTokenId: swapRequest.toTokenId
+          }
+        })
+
+        // Show modal with OK button:
+        const name = getWalletName(finalToWallet)
+        const fromCurrencyCode = getCurrencyCode(
+          fromWallet,
+          swapRequest.fromTokenId
+        )
+        const toCurrencyCode = getCurrencyCode(
+          finalToWallet,
+          swapRequest.toTokenId
+        )
+        const template = isWalletCreated
+          ? lstrings.ss_same_address_upgrade_created_3s
+          : lstrings.ss_same_address_upgrade_selected_3s
+        await Airship.show<string | undefined>(bridge => (
+          <ButtonsModal
+            bridge={bridge}
+            title={lstrings.exchange_generic_error_title}
+            message={sprintf(template, fromCurrencyCode, toCurrencyCode, name)}
+            buttons={{ ok: { label: lstrings.string_ok_cap } }}
+          />
+        ))
+        return
+      } catch (e) {
```

Comment:
Doesn't ESLint require this to be `(e: unknown)`?

Reply:
Please fix still?

Reply:
I didn't get any linter warnings here...

Reply:
`@typescript-eslint/use-unknown-in-catch-callback-variable`, applies to `Promise .catch(...)` callback parameters, not `try { ... } catch (e) { ... }`. It will not flag a try/catch clause.

The TypeScript option `useUnknownInCatchVariables: true` (default) only changes the inferred type of the catch variable to unknown. It does not require an explicit `: unknown `annotation and does not warn at the catch site. Errors will appear only where you use e unsafely (for example, e.message without narrowing), not at catch (e).